### PR TITLE
Fixes/#288 finish e gon100 re gas grid fails

### LINKED ISSUE #288: Finish-eGon100RE-gas-grid-fails
The task gas_grid.insert-gas-data is failing when scenario is eGon100RE.  Can only be merged after https://github.com/openego/powerd-data/pull/122

### PR DESCRIPTION
**Pull request by CarlosEpia**
Fixes #288

### CODE REVIEW

File: src/egon/data/datasets/scenario_parameters/parameters.py [hunk 1]
```diff
@@ -786,7 +786,7 @@ def gas(scenario):
         interest_rate = 0.07  # [p.u.]
 
         parameters = {
-            "main_gas_carrier": "H2",
+            "main_gas_carrier": "CH4",
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/scenario_parameters/parameters.py]: Could you explain why you changed the main_gas_carrier for eGon100RE?

Comment (CarlosEpia) [reply src/egon/data/datasets/scenario_parameters/parameters.py]: Thanks for the comment. It is a mistake. It is changed in aa1d9f22d8212a93377e3ae732f87b184e5005eb.

File: src/egon/data/datasets/gas_grid.py [hunk 2]
```diff
@@ -847,93 +863,25 @@ def insert_gas_data():
...
+    scenarios = []
+    if "eGon2035" in s:
+        scenarios.append("eGon2035")
+    if "eGon100RE" in s:
+        scenarios.append("eGon100RE")
 
-    insert_CH4_nodes_list(gas_nodes_list)
-    abroad_gas_nodes_list = insert_gas_buses_abroad()
+    for scn_name in scenarios:
+        download_SciGRID_gas_data()
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_grid.py]: Is it needed to download the data for each scenario?

File: src/egon/data/datasets/gas_grid.py [hunk 3]
```diff
@@ -318,25 +315,31 @@ def insert_gas_buses_abroad(scn_name="eGon2035"):
...
     gdf_abroad_buses["bus_id"] = range(new_id, new_id + len(gdf_abroad_buses))
 
-    # Add central bus in Russia
+    #Add central bus in Russia
     gdf_abroad_buses = pd.concat(
         [
             gdf_abroad_buses,
             pd.DataFrame(
-                index=[gdf_abroad_buses.index.max() + 1],
+                index=["RU"],
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_grid.py:335]: I did not test it, but I am wondering if text entries in the index are fine or might lead to problems.

Comment (CarlosEpia) [reply src/egon/data/datasets/gas_grid.py:335]: The index of gdf_abroad_buses is now based on strings. That is why I changed it directly to "RU". At the end, this value is neglected [here](https://github.com/openego/powerd-data/blob/be1b2ff6e7157b9bf3825023f15f6668fa4399b4/src/egon/data/datasets/gas_grid.py#L348-L365), so it is only important that it creates no errors like the one you described (str + int).

File: src/egon/data/datasets/gas_grid.py (hunk 2 above)
Comment (CarlosEpia) [reply src/egon/data/datasets/gas_grid.py]: Right. We need to execute download_SciGRID_gas_data() just once. Change applied in f46ed0c800ef188c0821a7f7e34c1fccd7d927d1

File: src/egon/data/datasets/gas_grid.py [hunk 4]
```diff
@@ -464,6 +467,46 @@ def insert_gas_pipeline_list(
...
+        gas_pipelines_list["bus1"].str.contains("N_88_NS_LMGN"), "country1"
+    ] = "'XX']"
+
+    gas_pipelines_list["country_code"] = gas_pipelines_list.apply(
+        lambda x: x["country0"] + "," + x["country1"], axis=1
+    )
+    gas_pipelines_list.drop(
+        columns=["bus0", "bus1", "country0", "country1"], inplace=True
+    )
+
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_grid.py:509]: Do you know why these changes were not needed before?

Comment (CarlosEpia) [reply src/egon/data/datasets/gas_grid.py:509]: I think it was necessary but since it does not lead to an error, but to a warning, it was probably ignored.

**Review by ClaraBuettner — APPROVED** [review]
